flocker/control/_persistence.py

```diff
@@ -0,0 +1,92 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Persistence of cluster configuration.
+"""
+
+from pickle import dumps, loads
+
+from twisted.application.service import Service
+from twisted.internet.defer import succeed
+
+from ._model import Deployment
+
+
+# These should not use Pickle!@!
+# https://clusterhq.atlassian.net/browse/FLOC-1241x
+def serialize_deployment(deployment):
+    """
+    Convert a ``Deployment`` object to ``bytes``.
+
+    :param Deployment deployment: Object to serialize.
+
+    :return bytes: Serialized object.
+    """
+    return dumps(deployment)
+
+
+def deserialize_deployment(data):
```

Comment:
Do we want to do any version checking here? I know we strongly suggest we don't provide compatability, but I'm certain that people will try anyway. It would be nice to error instead of carrying on with possibly invalid data.